### PR TITLE
Initial sync can use model methods

### LINKED ISSUE #55: InitialSyncJob always uses `find_collection_id` from API instead of calling `webflow_collection_id` on model
The InitialSyncJob always uses the `find_collection_id` API method (`initial_sync_job.rb:10`) instead of calling `webflow_collection_id` on the appropriate model. 

It would be nice if the `InitialSyncJob` would call the model for the `webflow_collection_id`, so any overwrite can be used, for example when the `collection_id` is set directly, without using the `slug`.

I wanted to create a PR for this, couldn't get the `bundle` working as I had errors with making `nio4r`. Will create a PR once fixed, but for now adding an issue as well.

### PR DESCRIPTION
**Pull request by aedificator-nl**
Fixes #55 

The initial sync job now uses `should_skip_webflow_sync?` and `webflow_collection_id` methods from the model.

This way any adjustments to those methods (for example hardcoding a `webflow_collection_id` instead of using the `slug`) are used in the `initial_sync_job` as well.

Also includes a Rails upgrade. This does not seem to break anything, but was needed to get it working locally because of a lagging `nio4r` version.

### CODE REVIEW

File: spec/models/concerns/webflow_sync/callbacks_spec.rb [hunk 1]
```diff
@@ -54,7 +54,7 @@
 
     context "when it shouldn't skip Webflow sync", vcr: { cassette_name: 'webflow_sync/should_skip_webflow_sync' } do
       it 'returns false' do
-        expect(article.send(:should_skip_webflow_sync?)).to be false
+        expect(article.should_skip_webflow_sync?).to be false
```
Comment (vfonic) [inline spec/models/concerns/webflow_sync/callbacks_spec.rb:57]: Thanks! 👍

**Review by vfonic — APPROVED** [review]
Impeccable! Thank you!

Let me see how I can have GitHub run the build and I'll release this as patch x.x.X version update. I think this could be considered as a bug fix rather than a feature addition.

Let me know if you believe differently.
I'll also try to upgrade to the newly released gem version from one of my Rails apps which are not on Rails 7.1.x to see if Bundler will try to update and fail or will it stay on the older version until I upgrade to Rails 7.1.x.

**Comment by vfonic** [comment]
Can you please have a look and fix the build issue(s)?

**Review by vfonic — CHANGES_REQUESTED** [review]
Looks like your RubyMine IDE files got checked in as well. Can you please remove them from the PR?

I have this in my `~/.gitignore_global`:

```
*~
.DS_Store
.idea/*
.pry_history
.vscode/*

.yarn/*
!.yarn/patches
!.yarn/releases
!.yarn/plugins
!.yarn/sdks
!.yarn/versions
.pnp.*

/TAGS

*.code-workspace
TODO
```

I hope it helps.

**Comment by aedificator-nl** [comment]
I have fixed some issues. The current issues with the test seem to have to do with the env vars, in test I think `WEBFLOW_SITE_ID` is set to `<WEBFLOW_SITE_ID>`, which errors out the tests. Could you replace this with an allowed dummy id?

**Comment by vfonic** [comment]
I don't know how to allow PRs from forks to share Github Actions secrets. I tested your changes both locally and I pushed it to a branch on this repo and the build passed.

I released this as v9.0.0 due to many gems being updated due to security issues.
I don't mind bumping major version number (as you can tell) than breaking someone's app because it feel like the changes don't justify a major version bump.

**Comment by aedificator-nl** [comment]
Thanks! Great that it all worked out!

**Review by sins921 — APPROVED** [review]

**Comment by vfonic** [comment]
WTF just happened here? :) How was this PR merged automatically?

**Comment by vfonic** [comment]
@aedificator-nl seems like the branch in your fork was deleted? https://github.com/aedificator-nl/webflow_sync/branches

https://github.com/aedificator-nl/webflow_sync/tree/feature/initial-sync-should-use-model-settings

That caused the false alarm that this was already merged, while it wasn't.

**Comment by aedificator-nl** [comment]
I have restored the branch. As far as I can see you have merged the changes into your master branch on the 6th of June 2024, after which I deleted the branch.

**Pull request merged**